script/shared/agent.ts

```diff
@@ -0,0 +1,147 @@
+import path from "path";
+import fs from "fs";
+
+import { Config } from "./config";
+import { defaultContextPath } from "./constants";
+import { Context, ContextHook } from "./context";
+import { extractByte32AddrFromBech32 } from "./utils";
+import { getContractInfo } from "./wasm";
+
+export type HplAgentConfig = {
+  name: string;
+  domainId: string;
+  chainId: string;
+  protocol: "cosmos";
+  rpcUrls: { http: string }[];
+  grpcUrl: string;
+  canonicalAsset: string; // usually same as gas token
+  bech32Prefix: string; // hrp
+  gasPrice: {
+    amount: string;
+    denom: string;
+  };
+  contractAddressBytes: 32;
+  index: {
+    from: number;
+    chunk: number;
+  };
+  blocks: {
+    reorgPeriod: 0; // instant finality ⭐️
+  };
+
+  mailbox: string; // hexed
+  interchainGasPaymaster: string; // hexed
+  validatorAnnounce: string; // hexed
+  merkleTreeHook: string; // hexed
+  testRecipient: string; // hexed
+};
+
+export async function fromContext(
+  network: Config["networks"][number],
+  context: Context
+): Promise<HplAgentConfig> {
+  const toHex = (v: string) => `0x${extractByte32AddrFromBech32(v)}`;
```

Comment:
Sounds like this is a more general util function that should somewhere else?

Reply:
There's `viem.toHex` but i'm not sure it makes left padded hex string